guides-markdown/vps-setup.md

```diff
@@ -0,0 +1,129 @@
+
+`cd %userprofile%/.ssh`
+
+On Linux/Mac:
+
+`cd ~/.ssh`
+
+Copy the contents of the `id_rsa.pub` file and paste them into Digital Ocean's form.
+
+`ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCsYKEA5LZCDyMF+ZbrPWeVIYso0ZzpIZx9L7R+CGqMPo0mrSlYeeaPbP1btM/Wis4a81EaTM7Y5kkKqZ4XB/LnRWp415XVl5QdtGF2l5tgiy2ootVxEwdrH0lXyGFHEpOwHU6MYdYCd+bgpQwa291Q4bOUJhGxNZ07L/rMtZfWhWL
++YL+JpSajg/uonu+4YKuFETggGLIuK+piTD9dvjiaThwKtqiCh2dnqdHztRYk+OehJUcof3tFl9kSRUmh9MVI7pDaOxCJWRaU1dsn9YaUwRkIyOwESHqBdCE9ZDU4FzNItRh2dYY4ukGv2iRqZoTrjcB8UGJepI65aINKNvdj email@nomail.com`
+
+8. Choose a hostname for the droplet and click create.
+
+Digital Ocean will provision your server. Now is a good time to grab coffee.
+
+
+## Amazon AWS
+
+1. Create an account at https://aws.amazon.com
+2. Launch a new instance from the console https://aws.amazon.com/console
+3. Choose 'Ubuntu 16.04' and at least 2CPU 4GiB for hardware.
+4. Continue to add storage. Add 500GiB.
+
+### A Note on hardware requirements
+These hardware requirements are for a full node. When run in SPV node, you will not need the additional block storage. To run a full node you must maintain a complete history of all Bitcoin transactions. Without the full history the node is unable to validate transactions. If it cannot validate transactions, it cannot validate blocks or mine transactions to blocks.
+
+SPV Nodes use something called a bloom filter to maintain a smaller set of records. For example, if your Bitcoin address were `1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa` your SPV node will ask its network peers for all transactions with an address that starts with `1A1z`. The returned data set will contain extra transactions irrelevant to your wallet. But its storage requirements will be far smaller than a full node. Consequently, SPV Nodes can only verify their own transactions.
+
+5. Review and Launch
+6. Amazon will prompt you for an ssh keypair, download a new keypair.
+7. Launch Instances.
+8. View Instances
+
+Amazon will provision your server. Now is a good time to grab coffee.
+
+
+## Connecting to your server
+
+#### Digital Ocean
+
+`ssh Your.Server.Ip.Here`
+
+E.g.
+
+`ssh 178.62.124.90`
+
+Accept the RSA key, and you will be at the command line
+
+
+#### Amazon
+
+In the same directory as the private key you downloaded,
+
+```chmod 400 test.pem
+ssh -i "test.pem" yourname@yourinstance.amazonaws.com
+```
+
+E.g.
+
+`ssh -i "test.pem" ubuntu@ec2-18-219-26-103.us-east-2.compute.amazonaws.com`
+
+Accept the RSA key, and you will be at the command line
+
+`ubuntu@ip-172-31-7-194:~$ `
+
+## Setting up the environment
+The VPS is setup, from here the intructions will be the same regardless of VPS provider.
+
+1. Install NVM
+2. Install Node
+4. Install build essential
+5. Install python
+
+```
+$ curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.8/install.sh | bash
+$ source ~/.bashrc
+$ nvm install 9.2.1
+$ sudo apt-get install build-essential
+$ sudo apt-get install python
+```
+## Install Bcoin
+
```

Comment:
Interesting to use the examples to run. I actually hadn't thought of that. I think this would be nice to have but it would also be helpful (and more future proof) to give some instructions on describing the command line instructions. 

This should also include installing the bcoin module globally (e.g. `npm install -g bcoin --production`), passing arguments like which network you want to run, enabling indexing, and having an api-key. Not necessary but a nice to have would be running as a daemon (in the background) and then maybe some  sort of auto-restart in case of a crash using something like [forever](https://www.npmjs.com/package/forever) would be very helpful for people I think.